 AlarmManagerService: null check the alarm operation

If the operation is null it will lead to a crash.

diff --git a/services/core/java/com/android/server/AlarmManagerService.java b/services/core/java/com/android/server/AlarmManagerService.java
index c0a66d6..1267242 100644
--- a/services/core/java/com/android/server/AlarmManagerService.java
+++ b/services/core/java/com/android/server/AlarmManagerService.java
@@ -2727,14 +2727,18 @@ public void handleMessage(Message msg) {
                     // now trigger the alarms without the lock held
                     for (int i=0; i<triggerList.size(); i++) {
                         Alarm alarm = triggerList.get(i);
-                        try {
-                            alarm.operation.send();
-                        } catch (PendingIntent.CanceledException e) {
-                            if (alarm.repeatInterval > 0) {
-                                // This IntentSender is no longer valid, but this
-                                // is a repeating alarm, so toss the hoser.
-                                removeImpl(alarm.operation);
+                        if (alarm.operation != null) {
+                            try {
+                                alarm.operation.send();
+                            } catch (PendingIntent.CanceledException e) {
+                                if (alarm.repeatInterval > 0) {
+                                    // This IntentSender is no longer valid, but this
+                                    // is a repeating alarm, so toss the hoser.
+                                    removeImpl(alarm.operation);
+                                }
                             }
+                        } else {
+                            removeImpl(alarm.operation);
                         }
                     }
                     break;
